backend/src/main/java/de/aljoschanyang/backend/services/QuestionService.java

```diff
@@ -0,0 +1,37 @@
+package de.aljoschanyang.backend.services;
+
+import de.aljoschanyang.backend.exceptions.NoQuestionsException;
+import de.aljoschanyang.backend.models.Question;
+import de.aljoschanyang.backend.models.QuestionDTO;
+import de.aljoschanyang.backend.repositories.QuestionRepo;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class QuestionService {
+    private final QuestionRepo questionRepo;
+
+    private QuestionDTO convertToDTO (Question question) {
+        return QuestionDTO.builder()
+                .id(question.id())
+                .query(question.query())
+                .options(question.options())
+                .answer(question.answer())
+                .build();
+    }
+
+    public List<QuestionDTO> getQuestionsByTopicId(String topicId) throws NoQuestionsException {
```

Comment:
The same as above. You don't need to add the throws clause for unchecked exception
```suggestion
    public List<QuestionDTO> getQuestionsByTopicId(String topicId) {
```

Reply:
I changed it